{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "ename": "IndentationError",
     "evalue": "expected an indented block (<ipython-input-16-0c206812e2d7>, line 45)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;36m  File \u001b[0;32m\"<ipython-input-16-0c206812e2d7>\"\u001b[0;36m, line \u001b[0;32m45\u001b[0m\n\u001b[0;31m    counts[k] = [counts[k]]\u001b[0m\n\u001b[0m         ^\u001b[0m\n\u001b[0;31mIndentationError\u001b[0m\u001b[0;31m:\u001b[0m expected an indented block\n"
     ]
    }
   ],
   "source": [
    "from pymongo import MongoClient\n",
    "from itertools import chain\n",
    "import re\n",
    "\n",
    "\n",
    "def cleanhtml(raw_html):\n",
    "    cleantext = raw_html\n",
    "    if isinstance(raw_html, str):\n",
    "        cleanr = re.compile('<.*a>')\n",
    "        cleantext = re.sub(cleanr, '', raw_html)\n",
    "    return cleantext\n",
    "\n",
    "\n",
    "def cleandate(raw_html):\n",
    "    cleaneddate = raw_html\n",
    "    if isinstance(raw_html, str):\n",
    "        cleandt = re.compile(r'([0-3][0-9])\\.([0-9]{2})\\.([0-9]{4})')\n",
    "        cleaneddate = re.sub(cleandt, '', raw_html)\n",
    "    return cleaneddate\n",
    "\n",
    "\n",
    "client = MongoClient(\"mongodb://localhost:27017/\")\n",
    "db = client['ds_shipment_tracking']\n",
    "dhl_shipment_events = db.dhl_shipment_events\n",
    "\n",
    "res = dhl_shipment_events.find()\n",
    "\n",
    "\n",
    "def cleanstatussen(res):\n",
    "    t = list(map(lambda x: list(map(lambda y: y.get('status'), x.get('events'))), res))\n",
    "    c = list(chain.from_iterable(t))#stati\n",
    "    f = list(map(lambda x: cleanhtml(x), c))\n",
    "    cdt = list(map(lambda x: cleandate(x), f))\n",
    "    #s = set(cdt)\n",
    "\n",
    "    #cleanlist = []\n",
    "    counts = {}\n",
    "    for status in cdt:\n",
    "        if status in counts:\n",
    "            counts[status] = counts[status]+1\n",
    "        else:\n",
    "            counts[status] =1\n",
    "    print(counts)\n",
    "    for k, v in counts.items():\n",
    "        counts[k] = [counts[k]]\n",
    "\n",
    "    return counts\n",
    "\n",
    "\n",
    "    '''\n",
    "        if status is not None:\n",
    "            cleanlist.append(i)\n",
    "\n",
    "    sortedcl = sorted(cleanlist)\n",
    "\n",
    "    countlist = ((i, sortedcl.count(i)) for i in sortedcl)\n",
    "    number_of_status = set(countlist)\n",
    "    #print(number_of_status)\n",
    "    '''\n",
    "\n",
    "'''\n",
    "    for i in number_of_status:\n",
    "        print(i)\n",
    "        print('\\n')\n",
    "'''\n",
    "\n",
    "'''   \n",
    "    with open('Alle_Statis.txt', 'w') as f:\n",
    "        for i in sortedcl:\n",
    "            f.write(\"%s\\n\\n\" % i)\n",
    "'''\n",
    "\n",
    "counts = cleanstatussen(res)\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "import pandas as pd\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['Die Sendung wurde elektronisch angeku00FCndigt. Sobald die Sendung von uns bearbeitet wurde, erhalten Sie weitere Informationen.',\n",
       " 'Die Sendung wurde von DHL bearbeitet und wird fu00FCr den Weitertransport in die Region des Empfu00E4ngers vorbereitet.',\n",
       " 'Die Sendung ist in der Region des Empfu00E4ngers angekommen und wird im nu00E4chsten Schritt zur Zustellbasis transportiert.',\n",
       " 'Die Sendung wurde in das Zustellfahrzeug geladen.',\n",
       " 'Die Sendung wurde erfolgreich zugestellt.',\n",
       " 'Die Sendung befindet sich auf dem Weg zur Packstation.',\n",
       " 'Die Sendung liegt in der  zur Abholung bereit.',\n",
       " 'Der Empfu00E4nger hat die Sendung aus der PACKSTATION abgeholt.',\n",
       " 'Die Sendung konnte beim 2. Zustellversuch nicht zugestellt werden.',\n",
       " 'Die Sendung konnte nicht zugestellt werden und wird in die Filiale gebracht. Fu00FCr die Abholung ist neben der Benachrichtigung die Vorlage Ihres Ausweises sowie ggf. eine Vollmacht erforderlich.',\n",
       " 'Die Sendung wird zur Abholung in die  gebracht. Fu00FCr die Abholung ist neben der Benachrichtigung die Vorlage Ihres Ausweises sowie ggf. eine Vollmacht erforderlich.',\n",
       " 'Der Empfu00E4nger hat die Sendung in der Filiale abgeholt.',\n",
       " 'Die Sendung wird fu00FCr den Weitertransport vorbereitet.',\n",
       " 'Die Sendung wird fu00FCr die Verladung ins Zustellfahrzeug vorbereitet.',\n",
       " 'Die Sendung konnte nicht zugestellt werden und wird in eine Packstation gebracht.',\n",
       " 'Leider war eine Zustellung der Sendung nicht mu00F6glich, da der Empfu00E4nger nicht ermittelt werden konnte. Die Sendung geht zuru00FCck an den Absender.',\n",
       " 'Leider war eine Zustellung Ihrer Sendung heute nicht mu00F6glich. Wir versuchen es am nu00E4chsten Werktag.',\n",
       " 'Leider war eine Zustellung der Sendung heute nicht mu00F6glich. Mu00F6gliche Gru00FCnde: Abbruch der Zustelltour aufgrund von Krankheit, Unfall, u00DCberschreitung der Arbeitszeit. Wir versuchen es am nu00E4chsten Werktag.',\n",
       " 'Die Sendung befindet sich wie gewu00FCnscht auf dem Weg zur Filiale. Sie werden informiert, sobald die Sendung in der Filiale eingetroffen ist.',\n",
       " 'Die Sendung wurde zur Abholung in die  gebracht.',\n",
       " 'Die Zustellung der Sendung verzu00F6gert sich. Sie erfolgt voraussichtlich am nu00E4chsten Werktag.',\n",
       " 'Die Retouren-Sendung wurde an DHL u00FCbergeben und wird fu00FCr den Weitertransport in die Region des Empfu00E4ngers vorbereitet.',\n",
       " 'Die Sendung wurde dem Empfu00E4nger zugestellt.',\n",
       " 'Der Empfu00E4nger hat die Annahme der Sendung verweigert.',\n",
       " 'Die Sendung befindet sich in der Zustellbasis und wird fu00FCr die Auslieferung vorbereitet.',\n",
       " 'Leider war eine Zustellung der Sendung heute nicht mu00F6glich, da der Empfu00E4nger nicht ermittelt werden konnte.',\n",
       " 'Die Sendung wurde zuru00FCckgestellt. Die Zustellung erfolgt voraussichtlich am nu00E4chsten Werktag.',\n",
       " 'Die u00DCberweisung des Nachnahme-Betrags an den Zahlungsempfu00E4nger ist erfolgt.',\n",
       " 'Die Sendung wurde ausgeliefert.',\n",
       " 'Die Sendung wurde leider fehlgeleitet. Gegebenenfalls verzu00F6gert sich dadurch die Zustellung.',\n",
       " 'Die Sendung konnte leider nicht in die gewu00FCnschte Packstation eingestellt werden und wurde in eine Filiale weitergeleitet.',\n",
       " 'Der Empfu00E4nger hat die Sendung innerhalb der siebentu00E4gigen Lagerfrist nicht in der Filiale abgeholt.',\n",
       " 'Die Sendung konnte leider nicht in der Packstation eingestellt werden. Es erfolgt ein erneuter Zustellversuch am nu00E4chsten Werktag.',\n",
       " 'Die Sendung wird zur Abholung in die  gebracht. Die fru00FChestmu00F6gliche Uhrzeit der Abholung kann der Benachrichtigung entnommen werden.',\n",
       " 'Die Sendung wurde storniert.',\n",
       " 'Die geringfu00FCgig beschu00E4digte Sendung wurde nachverpackt und wird jetzt zum Empfu00E4nger transportiert',\n",
       " 'Der Abholauftrag wurde zur Durchfu00FChrung am nu00E4chsten Werktag an den Zusteller u00FCbermittelt.',\n",
       " 'Die Abholung wurde erfolgreich ausgefu00FChrt.',\n",
       " 'Die Sendung befindet sich auf dem Weg zum Empfu00E4nger.',\n",
       " 'Eine Filiale wurde als neue Lieferadresse gewu00E4hlt',\n",
       " 'Die Sendung wurde zum Weitertransport aus der PACKSTATION entnommen.',\n",
       " 'Die Zustellung an einen gewu00FCnschten Ablageort/Nachbarn wurde gewu00E4hlt.',\n",
       " 'Eine Packstation wurde als neue Lieferadresse gewu00E4hlt',\n",
       " 'Aufgrund eines Nachsendeauftrags des Empfu00E4ngers wird die Sendung an eine neue Empfu00E4ngeradresse gesandt.',\n",
       " 'Auf Wunsch des Empfu00E4ngers erfolgt ein 2. Zustellversuch.',\n",
       " 'Die Sendung wurde abgeholt.',\n",
       " 'Die Zustellung am gewu00FCnschten Liefertag  wurde gewu00E4hlt.',\n",
       " 'Ihr Paket wird bei uns gelagert und am gewu00FCnschten Liefertag zugestellt.',\n",
       " 'Die Sendung wurde leider beschu00E4digt und muss neu verpackt werden. Wir informieren Sie hier u00FCber den nu00E4chsten Schritt.',\n",
       " 'Aufgrund einer Beschu00E4digung der Umverpackung wurde die Sendung nachbearbeitet.',\n",
       " 'Der gewu00FCnschte Liefertag kann leider nicht hinterlegt werden.',\n",
       " 'Aufgrund einer Beschu00E4digung verzu00F6gert sich der Transport der Sendung.',\n",
       " 'Der Empfu00E4nger hat die Sendung innerhalb der neuntu00E4gigen Lagerfrist nicht in der PACKSTATION abgeholt.',\n",
       " 'Leider ist eine Zustellung der Sendung nicht mu00F6glich, da sie nicht unseren Versandbedingungen entspricht. Die Sendung wird an den Absender zuru00FCckgeschickt.',\n",
       " 'Die Sendung befindet sich auf dem Weg zum Absender.',\n",
       " 'Aufgrund einer Beschu00E4digung wurde die Sendung nachbearbeitet.',\n",
       " None,\n",
       " 'Aufgrund einer negativen Alterspru00FCfung konnte die Sendung nicht zugestellt werden.',\n",
       " 'Die Sendung liegt in der Filiale zur Abholung bereit. Der Empfu00E4nger wurde benachrichtigt.',\n",
       " 'Die Sendung wurde vom Absender in der Filiale eingeliefert.',\n",
       " 'Die Abholung war nicht erfolgreich, da keine Sendung zur Abholung vorhanden war.',\n",
       " 'Sendung mit besonderen Zustellvorgaben: Die Zustellvorgaben des Versenders konnten nicht erfu00FCllt werden.',\n",
       " 'Die Sendung wurde an eine Hauspoststelle weitergeleitet.',\n",
       " 'Die Sendung in Paketkasten-Anlage eingestellt',\n",
       " 'Es erfolgt eine Ru00FCcksendung auf Wunsch des Absenders.',\n",
       " 'Der Empfu00E4nger der Sendung hat eine Schadensanzeige in der Filiale aufgegeben.',\n",
       " 'Die Abholung war nicht erfolgreich, da die Sendung nicht versandfertig war.',\n",
       " 'Die Sendung wurde am Wunschort abgeholt.',\n",
       " 'Die Sendung konnte heute leider nicht zugestellt werden. Sie wird voraussichtlich am nu00E4chsten Werktag in die Postfiliale gebracht.',\n",
       " 'Die Sendung wurde aus der Packstation entnommen.']"
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Shape of passed values is (70, 1), indices imply (70, 70)",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mcreate_block_manager_from_blocks\u001b[0;34m(blocks, axes)\u001b[0m\n\u001b[1;32m   1650\u001b[0m                 blocks = [make_block(values=blocks[0],\n\u001b[0;32m-> 1651\u001b[0;31m                                      placement=slice(0, len(axes[0])))]\n\u001b[0m\u001b[1;32m   1652\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36mmake_block\u001b[0;34m(values, placement, klass, ndim, dtype, fastpath)\u001b[0m\n\u001b[1;32m   3094\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3095\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mklass\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mndim\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mndim\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mplacement\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mplacement\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3096\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/blocks.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, values, placement, ndim)\u001b[0m\n\u001b[1;32m     86\u001b[0m                 \u001b[0;34m'Wrong number of items passed {val}, placement implies '\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 87\u001b[0;31m                 '{mgr}'.format(val=len(self.values), mgr=len(self.mgr_locs)))\n\u001b[0m\u001b[1;32m     88\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Wrong number of items passed 1, placement implies 70",
      "\nDuring handling of the above exception, another exception occurred:\n",
      "\u001b[0;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-9-bfe5817610f2>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mpd\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mDataFrame\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfrom_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcounts\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mlist\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcounts\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mkeys\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0morient\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"index\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36mfrom_dict\u001b[0;34m(cls, data, orient, dtype, columns)\u001b[0m\n\u001b[1;32m   1136\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'only recognize index or columns for orient'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1137\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1138\u001b[0;31m         \u001b[0;32mreturn\u001b[0m \u001b[0mcls\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1139\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1140\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mto_numpy\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mNone\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;32mFalse\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, data, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    449\u001b[0m                 \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    450\u001b[0m                     mgr = init_ndarray(data, index, columns, dtype=dtype,\n\u001b[0;32m--> 451\u001b[0;31m                                        copy=copy)\n\u001b[0m\u001b[1;32m    452\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    453\u001b[0m                 \u001b[0mmgr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0minit_dict\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m{\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mdtype\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/construction.py\u001b[0m in \u001b[0;36minit_ndarray\u001b[0;34m(values, index, columns, dtype, copy)\u001b[0m\n\u001b[1;32m    165\u001b[0m         \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mmaybe_infer_to_datetimelike\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    166\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 167\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mcreate_block_manager_from_blocks\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mvalues\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mindex\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    168\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    169\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mcreate_block_manager_from_blocks\u001b[0;34m(blocks, axes)\u001b[0m\n\u001b[1;32m   1658\u001b[0m         \u001b[0mblocks\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'values'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mb\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mb\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mblocks\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1659\u001b[0m         \u001b[0mtot_items\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0msum\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mb\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mb\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mblocks\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1660\u001b[0;31m         \u001b[0mconstruction_error\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mtot_items\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mblocks\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m0\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mshape\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0maxes\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0me\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1661\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1662\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.6/site-packages/pandas/core/internals/managers.py\u001b[0m in \u001b[0;36mconstruction_error\u001b[0;34m(tot_items, block_shape, axes, e)\u001b[0m\n\u001b[1;32m   1689\u001b[0m         \u001b[0;32mraise\u001b[0m \u001b[0mValueError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"Empty data passed with indices specified.\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1690\u001b[0m     raise ValueError(\"Shape of passed values is {0}, indices imply {1}\".format(\n\u001b[0;32m-> 1691\u001b[0;31m         passed, implied))\n\u001b[0m\u001b[1;32m   1692\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1693\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mValueError\u001b[0m: Shape of passed values is (70, 1), indices imply (70, 70)"
     ]
    }
   ],
   "source": [
    "pd.DataFrame.from_dict(counts, columns=list(counts.keys()), orient=\"index\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.6.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
